### PR TITLE
feat: Add low level support for animation blending

### PR DESCRIPTION
**Pull request by Rhuagh**
Still requires the user to do all blend weight handling.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/570)

### CODE REVIEW

**Comment by Rhuagh** [comment]
bors try+

File: amethyst_animation/src/resources.rs [hunk 1]
```diff
@@ -13,13 +14,16 @@ pub trait AnimationSampling: Send + Sync + 'static {
     /// The interpolation primitive
     type Primitive: InterpolationPrimitive + Clone + Copy + Send + Sync + 'static;
     /// The channel type
-    type Channel: Clone + Hash + Eq + Send + Sync + 'static;
+    type Channel: Debug + Clone + Hash + Eq + Send + Sync + 'static;
```
Comment (Xaeroxe) [inline amethyst_animation/src/resources.rs]: Debug bound may not be necessary.

Comment (Rhuagh) [reply amethyst_animation/src/resources.rs]: I could put it here or on every place where the channel is used, it's required for the use in `AnimationCommand`

**Comment by Rhuagh** [comment]
I'll do a documentation pass

**Comment by Rhuagh** [comment]
I'm also gonna add a function on the sampling trait for blending config (disabling it etc)

**Comment by Rhuagh** [comment]
@Xaeroxe I have done a doc pass, and also added a way to disable animation blending on a per target component+channel basis

**Comment by torkleyy** [comment]
:lgtm: 

Also great to see some more docs!

---

Reviewed 3 of 10 files at r1, 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/570#-:-L5OgQb00Tk0P5vkWgWF:biadkaa)*

**Pull request merged**